lib/pdk/util/filesystem.rb

```diff
@@ -2,6 +2,7 @@ module PDK
   module Util
     module Filesystem
       def write_file(path, content)
+        raise ArgumentError, "#{path} is empty" unless content
```

Comment:
I would probably go a step further here. `content` is expected to be a string due to the `...encode` command later.

An empty string would be a valid content, but a `nil` value would not.

Reply:
Also this should be using the `gettext` format e.g.

```
raise PDK::CLI::ExitWithError, _(
          '%{message} Rename the file or exclude it from the package ' \
          'by adding it to the .pdkignore file in your module.',
        ) % { message: e.message }
```